pkg/apis/garden/v1beta1/types.go

```diff
@@ -723,9 +723,9 @@ type ShootStatus struct {
 	// LastOperation holds information about the last operation on the Shoot.
 	// +optional
 	LastOperation *gardencorev1alpha1.LastOperation `json:"lastOperation,omitempty"`
-	// LastError holds information about the last occurred error during an operation.
+	// LastErrors holds information about the last occurred error(s) during an operation.
 	// +optional
-	LastError *gardencorev1alpha1.LastError `json:"lastError,omitempty"`
+	LastErrors []gardencorev1alpha1.LastError `json:"lastErrors,omitempty"`
```

Comment:
Testing it locally didn't cause errors. I guess because both are optional the status update simply removes one and adds the other.

Reply:
The field name is changed, too `lastError` -> `lastErrors`, so they are completely different fields. For sure data will be lost. But I am not sure whether it should fail if this field is presented in the etcd, but is not defined in the API?

Reply:
It's simply ignored if it's in etcd but not in the API. However, wrt backwards compatibility I guess we can't do this either.

Reply:
You're right that both should be present. The data loss can have an effect on the shoot care controller which I didn't take into account: if there already exist a LastError and it is not added to the LastErrors array the following code will act as if there are no errors: [shoot_care_control.go:177](https://github.com/gardener/gardener/blob/1ed7cdbbbbeb739b91fffb4f5c7d586c61a2815d/pkg/controllermanager/controller/shoot/shoot_care_control.go#L177). 
Additionally the transition to the new array field for the gardener dashboard will not have to happen immediately after it is added.

Thanks for the review and @vpnachev for pointing me to that documentation.

Reply:
@vpnachev @rfranzke After thinking a bit about the backwards compatibility there could be the following issues:

If we keep only `LastErrors`(plural) in the internal representation of `ShootStatus` and both `LastError` and `LastErrors` in the versioned APIs: 
1. During conversion from internal to versioned to create `LastError`(singular) there is no way to construct the `LastError.Description` the same as it would have been before this change (unless we get it from the `LastOperation.Description`).
2. During updates which first retrieve the `ShootStatus` and then only change either `LastError` or `LastErrors`(plural) we have to use `LastError.LastUpdateTime` to decide which one to add to the internal representation and then save.